File: web/app/types/trade.ts
import type { ISODateString, Numeric } from './api'

export type TradeSide = 'BUY' | 'SELL'
export type TradeStatus = 'OPEN' | 'CLOSED' | 'CANCELLED'

export interface Trade {
  id: string
  symbol: string
  side: TradeSide
  quantity: Numeric
  entry_price: Numeric
  exit_price: Numeric | null
  pnl: Numeric
  return_pct: Numeric
  fees: Numeric
  slippage: Numeric
  strategy_id: string | null
  strategy_name: string | null
  agent_confidence: Numeric | null
  status: TradeStatus
  opened_at: ISODateString
  closed_at: ISODateString | null
  duration_seconds: number | null
}

export interface TradeDetail extends Trade {
  proposal_id: string | null
  decision_id: string | null
  order_ids: string[]
}

File: backend/app/schemas/trade.py
"""Trade read API schemas (completed round trips; populated from Phase 7 onward)."""

from __future__ import annotations

import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict

from app.models.enums import TradeSide


class TradeSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    symbol: str
    side: TradeSide
    quantity: Decimal
    entry_price: Decimal
    exit_price: Decimal | None
    pnl: Decimal
    fees: Decimal
    return_pct: Decimal
    strategy_id: uuid.UUID | None
    proposal_id: uuid.UUID | None
    order_id: uuid.UUID | None
    opened_at: datetime
    closed_at: datetime | None


class TradePageSchema(BaseModel):
    items: list[TradeSchema]
    total: int
    page: int
    page_size: int
